.github/workflows/docs.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and host documentation
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Build documentation
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+    - name: Checkout 
+      uses: actions/checkout@v3
+    - name: Install nightly toolchain
+      run: rustup toolchain install nightly 
+    - name: Build Documentation
+      run: cargo +nightly doc --no-deps --workspace -F full
+      env:
+        RUSTDOCFLAGS: "--cfg docsrs"
+        RUSTFLAGS: "--cfg docsrs"
```

Comment:
Would it make sense to add caching here and pin the version of the nightly compiler?

Reply:
Yes we could do that. But I don't see that much benefit here. Pinning would mean we have to change the pinned version every time a new Rust version is released (potentially) and since the workflow is only triggered when push to master happens the performance benefit is not that big. 

But sure I could pin

Reply:
Fair enough, let's not bother then!